runtimes/huggingface/tests/test_runtime.py

```diff
@@ -1,64 +1,170 @@
+                    datatype="BYTES",
+                    parameters=Parameters(content_type="hg_jsonlist", headers=None),
+                    data=TensorData(
+                        root=[
+                            b'{"score": 0.9869917035102844, "start": 12, "end": 18, "answer": "Seldon"}'
+                        ]
+                    ),
+                )
+            ],
+        )
+
+        pipeline_kind = QuestionAnsweringPipeline
+    elif runtime.hf_settings.task == "text-generation":
+        inference_request = InferenceRequest(
+            inputs=[
+                RequestInput(
+                    name="text_inputs",
+                    shape=[1],
+                    datatype="BYTES",
+                    data=["The capital of France is called"],
+                ),
+            ]
+        )
+
+        inference_response = InferenceResponse(
+            model_name="foo",
+            model_version=None,
+            id=None,
+            parameters=None,
+            outputs=[
+                ResponseOutput(
+                    name="output",
+                    shape=[1, 1],
+                    datatype="BYTES",
+                    parameters=Parameters(content_type="hg_jsonlist", headers=None),
+                    data=TensorData(
+                        root=[
+                            b'[{"generated_text": "The capital of France is called Vichy, which it\'s named after a Roman Catholic town.\\n\\n\\n\\n\\nMore than 50,000 Roman Catholic students attended the university this year.\\nLATEST STORIES\\nWife and child"}]'
+                        ]
+                    ),
+                )
+            ],
+        )
+
+        pipeline_kind = TextGenerationPipeline
+
+    return (
+        runtime,
+        inference_request,
+        inference_response,
+        pipeline_kind,
+    )
+
+
+async def test_load(future_runtime: test_inputs_outputs):
+    (
+        runtime,
+        _inference_request,
+        _inference_response,
+        pipeline_kind,
+    ) = future_runtime
 
-async def test_load(future_runtime: HuggingFaceRuntime):
-    runtime = future_runtime
     assert runtime.ready
-    assert isinstance(runtime._model, QuestionAnsweringPipeline)
 
+    assert isinstance(runtime._model, pipeline_kind)
+
+
+async def test_unload(future_runtime: test_inputs_outputs):
+    (
+        runtime,
+        _inference_request,
+        _inference_response,
+        _pipeline_kind,
+    ) = future_runtime
 
-async def test_unload(future_runtime: HuggingFaceRuntime):
-    runtime = future_runtime
     assert runtime.ready
 
     unloaded = await runtime.unload()
     assert unloaded
 
 
-async def test_infer(
-    future_runtime: HuggingFaceRuntime, inference_request: InferenceRequest
-):
-    runtime = future_runtime
-    res = await runtime.predict(inference_request)
-    pred = json.loads(res.outputs[0].data[0])
-    assert pred["answer"] == "Seldon"
-
-
-async def test_infer_multiple(
```

Comment:
**explain:** Multiple inputs is still tested. This was effectively a duplicate test since it used the same `InferenceRequest` fixture, albeit with the validation on each input. That validation still happens via Pydantic when creating the instance.